src/common/warnings.ts

```diff
@@ -0,0 +1,35 @@
+export enum SDSWarningTypes {
+  ButtonMissingSDSProps = "buttonMissingProps",
+  ChipDeprecated = "chipDeprecated",
+  MenuSelectDeprecated = "menuSelectDeprecated",
+}
+
+const SDS_WARNINGS = {
+  [SDSWarningTypes.ButtonMissingSDSProps]: {
+    hasWarned: false,
+    message:
+      "Warning: Buttons without sdsStyle or sdsType props will be deprecated.",
+  },
+  [SDSWarningTypes.ChipDeprecated]: {
+    hasWarned: false,
+    message: "Warning: <Chip /> will be deprecated and replaced with <Tag />",
+  },
+  [SDSWarningTypes.MenuSelectDeprecated]: {
+    hasWarned: false,
+    message:
+      "Warning: MenuSelect will be deprecated and replaced with <DropdownMenu />",
+  },
+};
+
+export const showWarningIfFirstOccurence = (warningType: SDSWarningTypes) => {
+  if (
+    !(<string>warningType in SDS_WARNINGS) ||
+    SDS_WARNINGS[warningType].hasWarned
+  )
+    return;
```

Comment:
Because of typescript, you can't actually call this method if the warning type isn't in `SDSWarningTypes`, but I think it's potentially possible that we add a warning type without creating the field and object in the `SDS_WARNINGS` object.

Reply:
I think it would be better to have it be more restrictive. What do you think? For example, it would guide developers towards adding the warning here, so other developers can see which types of warnings exist at a glance. (or maybe I'm missing the problem 😓)

Reply:
Besides this one comment, this PR lgtm.

Reply:
So I guess there isn't really a problem per se, but my comment was more to explain why this check is helpful.  But now that you mention it, it might be worth adding a warning that the warning type isn't defined, and adding that to the object to prevent it from showing over and over.